src/fsharp/FSharp.Build/FSharpEmbedResXSource.fs

```diff
@@ -134,7 +132,7 @@ module internal {1} =
                     if getBooleanMetadata "GenerateSource" false item then
                         let moduleName =
                             match item.GetMetadata("GeneratedModuleName") with
-                            | null -> Path.GetFileNameWithoutExtension(item.ItemSpec)
+                            | null | "" -> Path.GetFileNameWithoutExtension(item.ItemSpec)
```

Comment:
perhaps use string.IsNullOrWhitspace and if then else. the match looks a bit hokey here.

Reply:
yes would be better